assembly_uploader/assembly_manifest.py

```
@@ -113,64 +113,106 @@ def __init__(
+            manifest_path = os.path.join(self.upload_dir, f"{assembly_basename}_others_{assembly_alias}.manifest")
+        else:
+            assembly_basename = "_".join(runs_list)
+            manifest_path = os.path.join(self.upload_dir, f"{assembly_basename}.manifest")
         #   skip existing manifests
         if os.path.exists(manifest_path) and not self.force:
             logging.warning(
-                f"Manifest for {run_id} already exists at {manifest_path}. Skipping"
+                f"Manifest for {run_ids} already exists at {manifest_path}. Skipping"
             )
             return
         values = (
             ("STUDY", self.new_project),
             ("SAMPLE", sample),
-            ("RUN_REF", run_id),
-            ("ASSEMBLYNAME", run_id + "_" + assembly_alias),
+            ("RUN_REF", run_ids),
+            ("ASSEMBLYNAME", assembly_basename + "_" + assembly_alias),
             ("ASSEMBLY_TYPE", "primary metagenome"),
             ("COVERAGE", coverage),
             ("PROGRAM", assembler),
             ("PLATFORM", sequencer),
             ("FASTA", assembly_path),
             ("TPA", str(self.tpa).lower()),
         )
-        logging.info("Writing manifest file (.manifest) for " + run_id)
+        logging.info("Writing manifest file (.manifest) for " + run_ids)
         with open(manifest_path, "w") as outfile:
             for k, v in values:
                 manifest = f"{k}\t{v}\n"
                 outfile.write(manifest)
 
     def write_manifests(self):
         for row in self.metadata:
-            ena_query = EnaQuery(row["Run"], self.private)
-            ena_metadata = ena_query.build_query()
+            # collect sample accessions and instrument models from runs
+            sample_accessions = set()
+            instruments = set()
+            for run in row["Run"].split(","):
+                # TODO in theory private/non-private state can be different for runs in co-assembly
```

I think we should probably raise an error for this case to be honest - unless we have known cases of it that we absolutely must support?

---

+1 we shouldn't do co-assemblies of public and private. The management of release dates will be a bit of a nightmare :D

---

I don't see where I could trace it to throw exception... Private or non-private is already set for all assemblies in CSV at once. The only thing I can do is to add explanation to README

---

Ah yeah I see, clarifying in the README is sufficient. If you want to be super thorough you could add a bad authentication response exception here https://github.com/EBI-Metagenomics/assembly_uploader/blob/9e076597ffc94d0b4311959b9c551816bbe319f1/assembly_uploader/ena_queries.py#L95 :D

---

There is a screaming note now in README 👻

---

Can remove this comment so no one tries in the future ;)